packages/react-aria-components/src/Menu.tsx

```diff
@@ -152,27 +153,45 @@ export interface MenuProps<T> extends Omit<AriaMenuProps<T>, 'children'>, Collec
 
 function Menu<T extends object>(props: MenuProps<T>, ref: ForwardedRef<HTMLDivElement>) {
   [props, ref] = useContextProps(props, ref, MenuContext);
+  let state = useContext(ExternalMenuStateContext);
```

Comment:
We need to use a separate menu state context than the one that exists currently since a Submenu should render its own StandaloneMenu with associated TreeState. If we reused MenuStateContext here then a Submenu would just reuse its parent menu's state, thus rendering an exact copy of its children